website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,223 @@
+---
+title: How to use feature flags with AI
+slug: /feature-flag-tutorials/use-cases/ai
+---
+
+Many tech companies today are integrating generative AI with large language models (LLMs) into their products, often in the form of chatbots or content generators.
+
+The main way to interact with LLMs today is via a set of APIs, usually either OpenAI, Anthropic or aggregators like Groq. Most of these APIs have similar parameters, like:
+
+-   `model`: The model and the specific version of the model to use.
+-   `prompt`: The prompt to give to the LLM.
+
+Given how much these models vary in their capabilities, chances are you'll be testing multiple models, multiple versions of the same model, and multiple prompts.
+
+This is where feature flags are super useful. They allow you to easily switch between different configurations. In this tutorial, we’ll explore how to use feature flags with AI models.
+
+We'll start with a basic chatbot. I know building a chatbot is a total cliché at this point, but it's a small, self-contained example that everyone knows how to interact with. You can take this example and apply it to any other use case where you need to interact with an LLM via an API.
+
+First, clone the repo:
+
+```sh
+git clone https://github.com/alvinometric/feature-flags-ai
+cd feature-flags-ai
+```
+
+Install the dependencies:
+
+```sh
+npm install
+```
+
+Run the app:
+
+```sh
+npm run dev
+```
+
+You should now see a chatbot UI, like this:
```

Comment:
We might want to mention that an OpenAI API key should be set up before validating the assumption that the output will match the screenshot.